Stage2 build

kassane opened the pull request:
I had to change the allocator because it does not allow testing outside the block.
Thus being compatible with the future v0.10 (stage2 or stage3) release

kassane committed:
fix - build on stage 2 (zig v0.10 - nightly)

kassane committed:
change allocator & fix help

tw4452852 commented on src/client.zig:
```diff
@@ -65,7 +65,8 @@ fn usage(args: *c.fuse_args) void {
 
 pub fn main(argv: [][*:0]u8, is_test: bool) !void {
     defer log(.info, "client exit\n", .{});
-    const allocator = if (is_test) std.testing.allocator else std.heap.c_allocator;
```
We could use `builtin.is_test` instead. This will let us eliminate `is_test` function parameter both in `client.main` and `server.main`.

kassane committed:
more fix:
- replace `openDir` to `openIterableDir`
- fix Entry iterateDir
- import not detect root fields, replaced to `main.zig` import
- builtin version compare e is_test added

kassane commented:
I have not been able to resolve this during testing yet. I don't have much thread usage.
```bash
 error: expected 1 argument(s), found 2
            @call(.{}, f, args) catch |err| {
```

--- 

Oh my god!! :fearful: 
I pressed the wrong button and it closed.

tw4452852 commented:
Hi @kassane 

I have fixed the issues related to the stage2. Due to 0.10 release hasn't been announced, I will maintain a seperated `stage2` branch to track. So close this PR, thanks for your work!

tw4452852 commented:
FYI. `stage2` branch has landed `master`.

kassane commented:
Nice!
